src/Microsoft.AspNetCore.Mvc.Analyzers/AttributesShouldNotBeAppliedToPageModelAnalyzer.cs

```diff
@@ -0,0 +1,168 @@
+                var location = GetAttributeLocation(ref symbolAnalysisContext, attribute);
+
+                symbolAnalysisContext.ReportDiagnostic(Diagnostic.Create(
+                    DiagnosticDescriptors.MVC1003_RouteAttributesShouldNotBeAppliedToPageModels,
+                    location,
+                    attribute.AttributeClass.Name));
+            }
+        }
+
+        private static void ReportRouteDiagnostic(ref SymbolAnalysisContext symbolAnalysisContext, IMethodSymbol method, INamedTypeSymbol routeAttribute)
+        {
+            var attribute = GetAttribute(method, routeAttribute);
+            if (attribute != null)
+            {
+                var location = GetAttributeLocation(ref symbolAnalysisContext, attribute);
+
+                symbolAnalysisContext.ReportDiagnostic(Diagnostic.Create(
+                    DiagnosticDescriptors.MVC1002_RouteAttributesShouldNotBeAppliedToPageHandlerMethods,
+                    location,
+                    attribute.AttributeClass.Name));
+            }
+        }
+
+        private static void ReportFilterDiagnostic(ref SymbolAnalysisContext symbolAnalysisContext, IMethodSymbol method, INamedTypeSymbol filterAttribute)
+        {
+            var attribute = GetAttribute(method, filterAttribute);
+            if (attribute != null)
+            {
+                var location = GetAttributeLocation(ref symbolAnalysisContext, attribute);
+
+                symbolAnalysisContext.ReportDiagnostic(Diagnostic.Create(
+                    DiagnosticDescriptors.MVC1001_FiltersShouldNotBeAppliedToPageHandlerMethods,
+                    location,
+                    attribute.AttributeClass.Name));
+            }
+        }
+
+        private static AttributeData GetAttribute(ISymbol symbol, INamedTypeSymbol attributeType)
+        {
+            foreach (var attribute in symbol.GetAttributes())
+            {
+                if (attributeType.IsAssignableFrom(attribute.AttributeClass))
+                {
+                    return attribute;
+                }
+            }
+
+            return null;
+        }
+
+        private static Location GetAttributeLocation(ref SymbolAnalysisContext symbolAnalysisContext, AttributeData attribute)
+        {
+            var syntax = attribute.ApplicationSyntaxReference.GetSyntax(symbolAnalysisContext.CancellationToken);
+            return syntax?.GetLocation() ?? Location.None;
+        }
+
+        private class TypeCache
+        {
+            public TypeCache(Compilation compilation)
+            {
+                PageModelAttribute = compilation.GetTypeByMetadataName(SymbolNames.PageModelAttributeType);
+                IFilterMetadata = compilation.GetTypeByMetadataName(SymbolNames.IFilterMetadataType);
+                AuthorizeAttribute = compilation.GetTypeByMetadataName(SymbolNames.AuthorizeAttribute);
+                AllowAnonymousAttribute = compilation.GetTypeByMetadataName(SymbolNames.AllowAnonymousAttribute);
```

Comment:
Should we handle EnableCors & DisableCors too?

Reply:
Not super sure there's a very high incidence of applying CORS attribute in Razor Pages. The programming model isn't really suitable for CORS to start with no?

Reply:
Yeah, I thought so too. You can leave this as it is then.